pkg/hs/urbit-king/lib/Urbit/Vere/Eyre/Serv.hs

```diff
@@ -254,19 +255,38 @@ startServer
   -> Net.Socket
   -> Maybe W.Port
   -> TVar E.LiveReqs
+  -> IO ()
   -> RIO e ()
-startServer typ hos por sok red vLive = do
+startServer typ hos por sok red vLive onFatal = do
   envir <- ask
 
   let host = case hos of
         SHLocalhost -> "127.0.0.1"
         SHAnyHostOk -> "*"
 
+  let handler r e = do
+        when (isFatal e) $ do
+          runRIO envir $ logError $ display $ msg r e
+          onFatal
+        when (W.defaultShouldDisplayException e) $ do
+          runRIO envir $ logWarn $ display $ msg r e
+
+      isFatal e
+        | Just (IOError {ioe_type = ResourceExhausted}) <- fromException e
+        = True
+        | otherwise = False
+
+      msg r e = case r of
+        Just r  -> "eyre: failed request from " <> (tshow $ W.remoteHost r)
+                <> " for " <> (tshow $ W.rawPathInfo r) <> ": " <> tshow e
+        Nothing -> "eyre: server exception: " <> tshow e
+
   let opts =
         W.defaultSettings
           & W.setHost host
           & W.setPort (fromIntegral por)
-          & W.setTimeout (5 * 60)
+          & W.setTimeout 30
```

Comment:
This timeout might be a little too aggressive, but it'll be easy to tune it later.

Reply:
IIRC it's the timeout vere uses.